classes/check/hashlegacy.php

```diff
@@ -0,0 +1,87 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+/**
+ * Check class for security check API
+ *
+ * @package   tool_hashlegacy
+ * @author    Peter Burnett <peterburnett@catalyst-au.net>
+ * @copyright 2020 Catalyst IT
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+namespace tool_hashlegacy\check;
+
+use core\check\check;
+use core\check\result;
+
+defined('MOODLE_INTERNAL') || die();
+
+class hashlegacy extends check {
+
+    const WARNING_THRESHOLD = 0.2;
```

Comment:
I'd probably make this much simpler and say any old hash is a warning, even a single one. We just want them removed